app/hooks/StrapiAPIHooks.ts

```diff
@@ -0,0 +1,31 @@
+import useSWR from "swr"
+import fetcher from "utils/fetcher";
+import config from "../config";
+import { type Footer } from "models/Strapi";
+
+interface SWR {
+  error?: Error;
+  isLoading: boolean;
+}
+
+interface FooterData extends SWR {
+  data: Footer;
+}
+
+export function useFooterData(): FooterData {
+  const { data, error, isLoading } = useSWR(
```

Comment:
Footer actually might be a good case for [useSWRImmutable](https://swr.vercel.app/docs/revalidation#disable-automatic-revalidations), as will be other static data. If a hook builder like above seems reasonable, we might have a couple different types of builders that correspond to different cache/retrieval strategies.

Reply:
I think this is a good thing to keep in mind, but the footer data, especially the nav, will likely change as we're coming up to launch.